### PR TITLE
ipa-server-install: add --setup-kra option

### PR DESCRIPTION
**Pull request by MartinBasti**

### CODE REVIEW

File: ipaserver/install/server/replicainstall.py [hunk 1]
```diff
@@ -836,14 +836,12 @@ def install_check(installer):
             root_logger.debug('No IPA DNS servers, '
                               'skipping forward/reverse resolution check')
 
-        kra_enabled = remote_api.Command.kra_is_enabled()['result']
```
Comment (tomaskrizek) [inline ipaserver/install/server/replicainstall.py]: If I understand this correctly, before this patch, KRA would be installed automatically if it was present on the master server. With this patch, `--setup-kra` would have to be explicitly passed to `ipa-replica-install`.

Is this the case or am I missing something? Is this behaviour change intended?

File: ipaserver/install/server/install.py [hunk 2]
```diff
@@ -384,6 +384,8 @@ def install_check(installer):
     print("  * Create and configure an instance of Directory Server")
     print("  * Create and configure a Kerberos Key Distribution Center (KDC)")
     print("  * Configure Apache (httpd)")
+    if options.setup_kra:
```
Comment (tomaskrizek) [inline ipaserver/install/server/install.py]: Please update the man pages with `--setup-kra` option for both `ipa-server-install` and `ipa-replica-install`.

File: ipaserver/install/server/replicainstall.py (hunk 1 above)
Comment (stlaz) [inline ipaserver/install/server/replicainstall.py]: No, you don't understand it. KRA would not be installed. See `kra.py:install()` for details.

**Comment by HonzaCholasta** [comment]
NACK on the "KRA: run install and install_check only when KRA should be installed" commit. The end goal for all component installers is to make them isolated and handle their options themselves, so that they can be packaged separately (among other things). This commit takes the code in the opposite direction. Also it does not make the code more readable because it is inconsistent with the CA installer.

File: ipaserver/install/server/__init__.py [hunk 3]
```diff
@@ -159,7 +159,6 @@ def domain_level(self, value):
         None,
         description="configure a dogtag KRA",
     )
-    setup_kra = enroll_only(setup_kra)
```
Comment (HonzaCholasta) [inline ipaserver/install/server/__init__.py]: Why did you remove this line? It must be here, otherwise `--setup-kra` will appear in `ipa-replica-prepare` once it's ported to the install framework.

File: ipaserver/install/server/replicainstall.py (hunk 1 above)
Comment (MartinBasti) [inline ipaserver/install/server/replicainstall.py]: commit removed

File: ipaserver/install/server/install.py (hunk 2 above)
Comment (MartinBasti) [inline ipaserver/install/server/install.py]: done

File: ipaserver/install/server/__init__.py (hunk 3 above)
Comment (MartinBasti) [inline ipaserver/install/server/__init__.py]: I somehow thought that it will not work, reverted.

**Comment by MartinBasti** [comment]
Given that there is no time, I dropped commit you NACKed as it unneeded for this PR, but please note my disagreement about a way how `kra.py` handles `--setup-kra` option for the future release.

**Comment by dkupka** [comment]
master:

* 4006cbbc02c368ac9e5e3721613158decb34fd37 KRA: add --setup-kra to ipa-server-install
* 25fa2bb6c9fa1b498330b13c9a6116b646eb75ba tests: use --setup-kra in tests
